x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -373,12 +373,10 @@ public void analyzedPlan(
         }
 
         PreAnalyzer.PreAnalysis preAnalysis = preAnalyzer.preAnalyze(parsed);
-        var unresolvedPolicies = preAnalysis.enriches.stream().map(EnrichPolicyResolver.UnresolvedPolicy::from).collect(toSet());
-
```

Comment:
No need to eagerly transform Enrich -> UnresolvedPolicy here. 
This is going to be transformed later in `enrichPolicyResolver` but only if there are actual policies to resolve.